oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/IndexName.java

```diff
@@ -231,26 +231,37 @@ public String nextCustomizedName() {
     }
 
     public static boolean isIndexActive(String indexPath, NodeState rootState) {
+        return isIndexActive(indexPath, rootState, 0);
+    }
+
+    private static boolean isIndexActive(String indexPath, NodeState rootState, int recursionDepth) {
         // An index is active if it has a hidden child node that starts with ":oak:mount-",
         // OR if it is an active merged index
-        try {
-            NodeState indexNode = rootState;
-            for (String e : PathUtils.elements(indexPath)) {
-                indexNode = indexNode.getChildNode(e);
+        if (recursionDepth > 50) {
+            LOG.warn("Fail to check index activeness for {} due to high recursion depth", indexPath);
```

Comment:
Log the recursion depth. And could this happen because of cycles in the index definition? We could explain in the log messages what are the possible causes and fixes for this error.

Reply:
Logging the recursion depths seems only needed if we make it configurable. But OK, it doesn't hurt.